redash/query_runner/qubole.py

```diff
@@ -67,8 +68,12 @@ def run_query(self, query, user):
                        api_url='%s/api' % self.configuration['endpoint'])
 
         try:
-            cls = PrestoCommand if(self.configuration['query_type'] == 'presto') else HiveCommand
-            cmd = cls.create(query=query, label=self.configuration['cluster'])
+            if self.configuration['query_type'] == 'quantum':
```

Comment:
This is an issue we had even before, but worth addressing here: `self.configuration['query_type']` might not have a value at all. The default value we mention is up to Query Runner implementation to enforce.

I would change this part of the code to something like:

```python
query_type = self.configuration.get('query_type', 'hive')

if query_type == 'quantum':
    cmd = SqlCommand.create(query=query)
elif query_type == 'hive':
   cmd = HiveCommand(query=query, label=self.configuration['cluster'])
elif query_type == 'presto':
   cmd = PrestoCommand(query=query, label=self.configuration['cluster'])
else:
   raise Exception("Invalid value for Query Type: {}. Should be one of: hive, presto, quantum.")
```

(this also adds handling of the case we get a bad value for query_type).

Reply:
I see none of the fields in `self.configuration` are getting populated with input, except for secrete property (token).
@arikfr i am witnessing following error, on executing first query using registered data source.Any suggestions?
Code:
```
query_type = self.configuration.get('query_type')
 if query_type == 'quantum':
     cmd = SqlCommand.create(query=query)
 elif query_type == 'hive':
     cmd = HiveCommand.create(query=query, label=self.configuration.get('cluster'))
 elif query_type == 'presto':
     cmd = PrestoCommand.create(query=query, label=self.configuration.get('cluster'))
 else:
     raise Exception("query_type:%s\n end_point:%s\n cluster:%s\n token:%s\n"%(self.configuration.get('query_type'),self.configuration.get('end_point'), self.configuration.get('cluster'), self.configuration.get('token')))
```

Error:

>  Traceback (most recent call last):
worker_1    |   File "/app/redash/tasks/queries.py", line 340, in run
worker_1    |     data, error = query_runner.run_query(annotated_query, self.user)
worker_1    |   File "/app/redash/query_runner/qubole.py", line 82, in run_query
worker_1    |     self.configuration.get('end_point'), self.configuration.get('cluster'), self.configuration.get('token')))
worker_1    | Exception: query_type:None
worker_1    |  end_point:None
worker_1    |  cluster:None
worker_1    |  token:3b2febd544exxxxxxxxx>

Reply:
You didn't provide a default value to `query_type`, change the first line to:

```python
query_type = self.configuration.get('query_type', 'hive')
```

Reply:
Done, qubole.py module was missing class method for `type` and `name`. This lead to failure of picking right object name. Things worked for me after adding them.